packages/xml-views-validation/src/validate-xml-views.ts

```diff
@@ -0,0 +1,55 @@
+import { flatMap } from "lodash";
+import {
+  XMLDocument,
+  XMLAstVisitor,
+  XMLAttribute,
+  XMLElement,
+  accept
+} from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { UI5XMLViewIssue } from "../api";
+
+export interface UI5Validators {
```

Comment:
Do you think there might be a need for validators that use a visitor to check that something exists in the document? Not just in relation to a specific node (like that associations point to an existing element, or if a namespace declaration is unused - these should be in attribute value validators but will have to use a visitor or other means of finding something in the xml), but in general? I don't have a use case right now... maybe it's more something to consider when we have a use case.

Reply:
I think there would be such scenarios, but it may be possible to model these with the existing API, e.g:
- unused namespace warning:
  - Start at attribute which is `xmlns`
  - go to the parent XMLDocument (.parent.parent.parent...parent)
  - Start a new Visitor for tags only to see a use.

Obviously this is not the most efficient, but it logically keeps the different validations very separated / de-coupled. I would like to see an actual performance issues before trying to optimize it (which may lose the separation).

Another possibility is to run a validator on every tag that only collects and fill some state, (e.g maps of all used + declared xmlns prefixes), and then run a post traversal validation on the collected state, that would be more efficient, and also keep the separation.

Seems easy enough to expend the API when we will need it...